src/modes/map/map_mode.cpp

```diff
@@ -968,46 +978,53 @@ void MapMode::_UpdateMapFrame()
     _map_frame.num_draw_x_axis = TILES_ON_X_AXIS + 1;
     _map_frame.num_draw_y_axis = TILES_ON_Y_AXIS + 1;
 
-    // Camera exceeds the left boundary of the map
-    if(_map_frame.tile_x_start < 0) {
-        _map_frame.tile_x_start = 0;
+    int32_t width_offset = static_cast<int32_t>(vt_video::VIDEO_STANDARD_RES_WIDTH / vt_map::private_map::SCREEN_GRID_X_LENGTH / vt_map::private_map::MAP_ZOOM_RATIO);
+    float width_offset_screen = width_offset * vt_map::private_map::MAP_ZOOM_RATIO;
+
+    // Camera exceeds the left boundary of the map.
+    if (_map_frame.tile_x_start < -width_offset) {
+        _map_frame.tile_x_start = -width_offset;
         _map_frame.tile_x_offset = vt_utils::FloorToFloatMultiple(1.0f, _pixel_length_x);
-        _map_frame.screen_edges.left = 0.0f;
-        _map_frame.screen_edges.right = SCREEN_GRID_X_LENGTH;
+        _map_frame.screen_edges.left = -width_offset_screen;
+        _map_frame.screen_edges.right = SCREEN_GRID_X_LENGTH - width_offset_screen;
         _map_frame.num_draw_x_axis = TILES_ON_X_AXIS;
         _camera_x_in_map_corner = true;
     }
-    // Camera exceeds the right boundary of the map
-    else if(_map_frame.tile_x_start + TILES_ON_X_AXIS >= _tile_supervisor->_num_tile_on_x_axis) {
-        _map_frame.tile_x_start = static_cast<int16_t>(_tile_supervisor->_num_tile_on_x_axis - TILES_ON_X_AXIS);
-        _map_frame.tile_x_offset = vt_utils::FloorToFloatMultiple(1.0f, _pixel_length_x);
-        _map_frame.screen_edges.right = static_cast<float>(_object_supervisor->_num_grid_x_axis);
-        _map_frame.screen_edges.left = _map_frame.screen_edges.right - SCREEN_GRID_X_LENGTH;
+    // Camera exceeds the right boundary of the map.
+    else if (_map_frame.tile_x_start + TILES_ON_X_AXIS >= _tile_supervisor->_num_tile_on_x_axis + width_offset) {
+        _map_frame.tile_x_start = static_cast<int16_t>(_tile_supervisor->_num_tile_on_x_axis - TILES_ON_X_AXIS + width_offset);
+        _map_frame.tile_x_offset = vt_utils::CeilToFloatMultiple(1.0f, _pixel_length_x);
```

Comment:
I think this change is alright.  On the upper boundary cases, "ceil" correcting pushes the value towards the upper boundary.